Brainfuck/R/bf.R

```diff
@@ -0,0 +1,59 @@
+library(stringr)
+
+args = commandArgs(trailingOnly=TRUE)
+
+program <- strsplit(readLines(args[1]), "")[[1]]
+
+dataStack <- c(rep(0, 30000))
+dataPointer <- 1
+
+instructionStack <- c()
+instructionPointer <- 1
+
+while (instructionPointer < (length(program) + 1)) {
+  switch (program[instructionPointer],
+    '>' = dataPointer <- dataPointer + 1,
+    '<' = dataPointer <- dataPointer - 1,
+    '+' = dataStack[dataPointer] <- dataStack[dataPointer] + 1,
+    '-' = dataStack[dataPointer] <- dataStack[dataPointer] - 1,
+    '.' = cat(rawToChar(as.raw(dataStack[dataPointer]))),
+    ',' = {
+      lines <- readLines(con = file("stdin"))

```

Comment:
Since you're splitting after reading isn't there a chance you are reading too much and throwing some away? Nothing should be thrown away. Only 1 byte should be read.

Reply:
Yeah this is definitely not correct.

You should probably use https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/readChar instead?

Reply:
Sorry about that. This was placeholder code I forgot to go back to. I'll fix it today if I get the time.

Reply:
Done